### PR TITLE
feat(build-lib): Update iob_uart_setup.py; Add source copy target.

### PR DESCRIPTION
**Pull request by arturum1**
- Add hw_setup.mk target to copy UART sources
- Add macros
- Comment non PIO signals temporarily to prevent them from appearing in iob-soc top module

### CODE REVIEW

**Comment by jjts** [comment]
why are checks failing?

File: iob_uart_setup.py [hunk 1]
```diff
@@ -9,30 +9,36 @@
...
-    {'name':'ADDR_W', 'type':'P', 'val':'`IOB_UART_SWREG_ADDR_W', 'min':'1', 'max':'NA'},
-    {'name':'UART_DATA_W', 'type':'P', 'val':'8', 'min':'1', 'max':'NA'}
+    # Macros
+    {'name':'DATA_W', 'type':'M', 'val':'32', 'min':'1', 'max':'NA', 'descr':"Data bus width"},
+    {'name':'ADDR_W', 'type':'M', 'val':'`IOB_UART_SWREG_ADDR_W', 'min':'1', 'max':'NA', 'descr':"Address bus width"},
+    {'name':'UART_DATA_W', 'type':'M', 'val':'8', 'min':'1', 'max':'NA', 'descr':""},
+    # Parameters
+    {'name':'DATA_W', 'type':'P', 'descr':"Data bus width"},
+    {'name':'ADDR_W', 'type':'P', 'descr':"Address bus width"},
+    {'name':'UART_DATA_W', 'type':'P', 'descr':""}
```
Comment (jjts) [inline iob_uart_setup.py:19]: Repeat?

**Review by jjts — COMMENTED** [review]
Check comments

File: iob_uart_setup.py (hunk 1 above)
Comment (arturum1) [reply iob_uart_setup.py:19]: The default values of parameters are auto generated by setup.py
For example, the parameter 'ADDR_W' will have the default auto generated value '`IOB_UART_ADDR_W'
The macro 'IOB_UART_ADDR_W' (just 'ADDR_W' in the configuration file) must be defined to assign the default value of the parameter.

Comment (jjts) [reply iob_uart_setup.py:19]: UART_DATA_W appears twice

Comment (arturum1) [reply iob_uart_setup.py:19]: One of them is a macro, the other is a parameter. The macro sets the value, the parameter uses the macro as its default value.

Comment (arturum1) [reply iob_uart_setup.py:19]: Note: 'DATA_W' and 'UART_DATA_W' are different macros/parameters.

**Pull request merged**